website/addons/googledrive/model.py

```diff
@@ -111,218 +101,183 @@ def get_or_create(cls, node, path):
         return new, created
 
 
-class GoogleDriveOAuthSettings(StoredObject):
-    """
-    this model address the problem if we have two osf user link
-    to the same google drive user and their access token conflicts issue
-    """
+class GoogleDriveProvider(ExternalProvider):
+    name = 'Google Drive'
+    short_name = 'googledrive'
+
+    client_id = drive_settings.CLIENT_ID
+    client_secret = drive_settings.CLIENT_SECRET
+
+    auth_url_base = "https://accounts.google.com/o/oauth2/auth?access_type=offline&approval_prompt=force"
```

Comment:
It could be broken up and rebuilt when it's needed by the superclass, but it will always be the same. It should probably be stored in settings, however.